opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/SLUB.kt

```diff
@@ -159,19 +159,26 @@ open class SLUB : OkHttpBaseApi() {
 
     override fun getResultById(id: String, homebranch: String?): DetailedItem {
         val json: JSONObject
+        val url =
+                if (id.startsWith("id")) {
+                    "$baseurl/$id/"
+                } else {
+                    val noRedirectClient = http_client.newBuilder()
```

Comment:
Do you really need to do this separate HEAD request to capture the redirected URL? You could also simply to the GET request (which should follow the redirect) and then use that as your JSON body directly, right?

Reply:
Yes, the redirect leads to the web catalog view, i.e. it returns a html, not a json. In order to get the json for the redirected site you need to add `?type=1369315142&tx_find_find[format]=data&tx_find_find[data-format]=app`. 
Of course all this could be done using an interceptor, but I found it easier to implement and test it this way.

Reply:
Ah, okay, yes, if the URL of the redirect needs to be changed that is probably the easiest solution.